### ISSUE TITLE
Mac sendfile 下载二进制文件后下载的文件已损坏

### ISSUE THREAD

**Issue by Awezome** [swoole/swoole-src#1966]
在http server里使用sendfile下载文件，如果下载的是二进制文件，下载后文件会损害。

测试代码：

```<?php

$http=new swoole_http_server('0.0.0.0',8811);

$http->set([
    'buffer_output_size' => 100 * 1024 *1024, //必须为数字
    'package_max_length' => 8192, //必须为数字
    'http_compression' => false,
]);

$http->on('request',function($request,$response){
    $filename='/Users/z/Desktop/1.dmg';
    $response->header('Content-Type','application/octet-stream',true);
    $response->sendfile($filename);
});

$http->start();
```

版本信息
mac os 10.13.6

```
$ php -v
PHP 7.2.9 (cli) (built: Aug 23 2018 02:08:27) ( NTS )
Copyright (c) 1997-2018 The PHP Group
Zend Engine v3.2.0, Copyright (c) 1998-2018 Zend Technologies
    with Zend OPcache v7.2.9, Copyright (c) 1999-2018, by Zend Technologies
```

```
$ php --ri swoole
swoole
swoole support => enabled
Version => 4.1.2
Author => Swoole Group[email: team@swoole.com]
coroutine => enabled
kqueue => enabled
rwlock => enabled
pcre => enabled
zlib => enabled

Directive => Local Value => Master Value
swoole.enable_coroutine => On => On
swoole.aio_thread_num => 2 => 2
swoole.display_errors => On => On
swoole.use_namespace => On => On
swoole.use_shortname => On => On
swoole.fast_serialize => Off => Off
swoole.unixsock_buffer_size => 8388608 => 8388608
```

**Comment by matyhtf** [comment]
使用 Linux 试试呢

**Comment by flddr** [comment]
@Awezome is it due `$response->header('Content-Disposition', 'attachment; filename="name.xyz"');`-header is missing?

Actually your file has no name and will be treated as client will do (without `.dmg`, because it is not known)

**Comment by Awezome** [comment]
> 使用 Linux 试试呢

使用linux可以

**Comment by Awezome** [comment]
> @Awezome is it due `$response->header('Content-Disposition', 'attachment; filename="name.xyz"');`-header is missing?
> 
> Actually your file has no name and will be treated as client will do (without `.dmg`, because it is not known)

The「Content-Disposition」setting is not necessary . We can rename the downloaded file to the right name if the file is available.

**Comment by twose** [comment]
Fixed via 757aafd, Thanks for feedback.

**Comment by twose** [comment]
Fixed via 757aafd, Thanks for feedback.
